client/layout/masterbar/item.jsx

```diff
@@ -1,46 +1,49 @@
 /**
  * External dependencies
  */
-import React from 'react';
+import React, { Component } from 'react';
 import classNames from 'classnames';
 import noop from 'lodash/noop';
 import isFunction from 'lodash/isFunction';
 import Gridicon from 'gridicons';
 
-export default React.createClass( {
-	displayName: 'MasterbarItem',
-
-	propTypes: {
-		url: React.PropTypes.string,
-		onClick: React.PropTypes.func,
-		tooltip: React.PropTypes.string,
-		icon: React.PropTypes.string,
-		className: React.PropTypes.string,
-		isActive: React.PropTypes.bool,
-		preloadSection: React.PropTypes.func
-	},
-
-	_preloaded: false,
-
-	getDefaultProps() {
-		return {
-			icon: '',
-			onClick: noop
-		};
-	},
+class MasterbarItem extends Component {
+	constructor() {
+		super();
+		this._preloaded = false;
```

Comment:
I guess that if this is moved outside the constructor as a class property, and the bind is replaced with an arrow function, we could drop the constructor altogether.

```es6
class MasterbarItem extends Component {
  _preloaded = false;

  preload = () => {
    ...
  }

  ...
}
```